Base/src/main/java/io/deephaven/base/clock/Clock.java

```diff
@@ -3,87 +3,58 @@
+     *
+     * @return epoch microseconds
+     */
     long currentTimeMicros();
 
-    interface Factory {
-        Clock getClock();
-    }
-
-    class Null implements Clock {
-        @Override
-        public long currentTimeMillis() {
-            return 0;
-        }
-
-        @Override
-        public long currentTimeMicros() {
-            return 0;
-        }
-    }
-
-    Null NULL = new Null();
-
     /**
-     * This implementation just returns the last value passed to set(). It allows for precise control over when clock
-     * calls are made to the underlying system (e.g. AppClock.currentTimeMicros()).
+     * Nanoseconds since the epoch, 1970-01-01T00:00:00Z.
+     *
+     * <p>
+     * Note: this value will overflow in the year 2292.
+     *
+     * @return epoch nanoseconds
      */
-    class Cached implements Clock {
-        private long cachedNowMicros;
-
-        public void set(long nowMicros) {
-            cachedNowMicros = nowMicros;
-        }
-
-        @Override
-        public final long currentTimeMillis() {
-            return cachedNowMicros / 1000;
-        }
-
-        @Override
-        public final long currentTimeMicros() {
-            return cachedNowMicros;
-        }
-    }
+    long currentTimeNanos();
 
     /**
-     * This implementation is similar to cached, except that is calls set() itself on a the Clock instance given to the
-     * constructor exactly once between reset() calls.
+     * The instant.
+     *
+     * @return the instant
      */
-    class CachedOnDemand implements Clock {
-        private final Clock realClock;
-        private long cachedNowMicros;
-
-        public CachedOnDemand(Clock realClock) {
-            this.realClock = realClock;
-        }
-
-        public void set() {
-            cachedNowMicros = realClock.currentTimeMicros();
-        }
+    Instant currentTimeInstant();
 
-        public void reset() {
-            cachedNowMicros = 0;
-        }
-
-        private long maybeUpdate() {
-            if (cachedNowMicros == 0) {
-                set();
-            }
-            return cachedNowMicros;
-        }
-
-        @Override
-        public final long currentTimeMillis() {
-            return maybeUpdate() / 1000;
-        }
-
-        @Override
-        public final long currentTimeMicros() {
-            return maybeUpdate();
-        }
-    }
+    /**
+     * Provides a nanosecond timer for measuring elapsed time. This may not be related to any notion of system or
+     * wall-clock time, so the results should only be compared with results from like-calls.
+     *
+     * <p>
+     * For example, to measure how long some code takes to execute:
+     * 
+     * <pre>
+     * long startNanoTime = clock.nanoTime();
+     * // ... the code being measured ...
+     * long elapsedNanos = clock.nanoTime() - startNanoTime;
+     * </pre>
+     * 
+     * @return the nano time
+     */
+    long nanoTime();
```

Comment:
No - this is explicitly required IMO when you want to have a replayer, but want to be able to finely replay "timings". For example, to use AutoTuningIncrementalReleaseFilter with a replayer, we need to rely on `Clock.nanoTime()` instead of `System.nanoTime()`.

Reply:
Ok, I'm still a bit skeptical of the legitimacy of case 1, but it does make the interface more complete